samples/hosting/generic-host/Core_7/GenericHost/Program.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using NServiceBus;
+
+internal class Program
+{
+    public static void Main(string[] args)
+    {
+        CreateHostBuilder(args).Build().Run();
+    }
+
+    private static IHostBuilder CreateHostBuilder(string[] args)
+    {
+        #region generic-host-lifetime
+        var builder = Host.CreateDefaultBuilder(args);
+        var isService = !(Debugger.IsAttached || args.Contains("--console"));
+        builder = isService ? builder.UseWindowsService() : builder.UseConsoleLifetime();
+        #endregion
+
+        #region generic-host-logging
+        builder.UseMicrosoftLogFactoryLogging();
+        builder.ConfigureLogging((ctx, logging) =>
+        {
+            logging.AddConfiguration(ctx.Configuration.GetSection("Logging"));
+
+            if (isService)
+                logging.AddEventLog();
+            else
+                logging.AddConsole();
+        });
+        #endregion
+
+        #region generic-host-nservicebus
+        builder.UseNServiceBus(ctx =>
+        {
+            var endpointConfiguration = new EndpointConfiguration("Samples.Hosting.GenericHost");
+
+            endpointConfiguration.UseTransport<LearningTransport>();
+            endpointConfiguration.UsePersistence<LearningPersistence>();
+
+            endpointConfiguration.DefineCriticalErrorAction(OnCriticalError);
+
+            return endpointConfiguration;
+        });
+        #endregion
+
+        #region generic-host-worker-registration
+        return builder.ConfigureServices(services =>
+        {
+            services.AddHostedService<Worker>();
+        });
+        #endregion
+    }
+    
+    #region generic-host-critical-error
+
+    private static async Task OnCriticalError(ICriticalErrorContext context)
+    {
+        var fatalMessage =
+            $"The following critical error was encountered:{Environment.NewLine}{context.Error}{Environment.NewLine}Process is shutting down. StackTrace: {Environment.NewLine}{context.Exception.StackTrace}";
+
+        EventLog.WriteEntry(".NET Runtime", fatalMessage, EventLogEntryType.Error);
+
+        if (Environment.UserInteractive)
```

Comment:
when running in VS, console apps don't close automatically anymore so this shouldn't be needed anymore? Not sure it's the same with FailFast though?

Reply:
Ok but what about if you run it with the console switch instead with the debugger attached?

Reply:
shouldn't we then ask for user input before closing too instead of just delaying?

Is this a bit too much functionality for a sample anyway?

Reply:
It was mentioned in the guidance so I figured I add it. I'm not opinionated enough to go one way or the other